### ISSUE TITLE
Why  bytes data returns null in v5-beta? 

### ISSUE THREAD

**Issue by rstormsf** [trufflesuite/truffle#1229]
in truffle v4 test would look like this:
solidity:
```
bytes public data;
```

```js
        (await erc677r.data()).should.be.equal('0x');
```

now in truffle v5, it returns `null`, not `0x`

**Comment by rstormsf** [comment]
Second issue is that events in ganache-cli does not emit if it was called by  
```
_to.call(abi.encodeWithSignature("onTokenTransfer(address,uint256,bytes)",  msg.sender, _value, _data));
```

**Comment by cgewecke** [comment]
Hi @rstormsf 

Yes - just saw the same decoding problem with `boolean` [here](https://github.com/trufflesuite/truffle/pull/1200) last week. Will investigate some more to see if it's on this side but I *think* it's at `web3-eth-abi` and might be resolved by a pending release over there. 

Will also look at problem #2. . . you're note suggests you think it's at ganache-cli - are you seeing this fire if you use `geth --dev` or do you mean that Truffle isn't detecting the event?
